pkg/scte35/stream.go

```diff
@@ -0,0 +1,354 @@
+// Copyright 2022 Adrian of Doom
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or   implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+package scte35
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+)
+
+const (
+    // pktSz is the size of an MPEG-TS packet in bytes.
+    pktSz = 188
+
+    // bufSz is the size of a read when parsing files.
+    bufSz = 13000 * pktSz
+)
+
+type PacketData struct {
+	PacketNumber int     `json:",omitempty"`
+	Pid          uint16  `json:",omitempty"`
+	Program      uint16  `json:",omitempty"`
+	Pcr          float64 `json:",omitempty"`
+	Pts          float64 `json:",omitempty"`
+}
+
+//Stream for parsing MPEGTS for SCTE-35
+type Stream struct {
+	pktNum   int // packet count.
+	programs []uint16
+	pid2Prgm map[uint16]uint16 //lookup table for pid to program
+	prgm2pcr map[uint16]uint64 //lookup table for program to pcr
```

Comment:
`programToPCR` here (not going to call out all these, just spotting some random examples and you can apply the same suggestions elsewhere.